src/dg/dg.cpp

```diff
@@ -3330,6 +3330,839 @@ real2 DGBase<dim,real,MeshType>::discontinuity_sensor(
+        real minmod = 0.0;
+
+        for (unsigned int istate = 0; istate < nstate; ++istate) {
+            a = soln_cell_avg[istate] - soln_cell_0[istate];
+            if (abs(a) <= M[istate] * pow(this->h, 2.0))
+            {
+                soln_0_lim[istate] = soln_cell_avg[istate] - a;
+            }
+            else //if (abs(a) > M * h * h)
+            {
+                if (signbit(a) == signbit(diff_next[istate]) && signbit(a) == signbit(diff_prev[istate]))
+                {
+                    minmod = std::min({ abs(a), abs(diff_next[istate]), abs(diff_prev[istate]) });
+
+                    if (signbit(a))
+                        soln_0_lim[istate] = soln_cell_avg[istate] + std::max(abs(minmod), M[istate] * pow(this->h, 2.0));
+                    else
+                        soln_0_lim[istate] = soln_cell_avg[istate] - std::max(abs(minmod), M[istate] * pow(this->h, 2.0));
+                }
+                else
+                    soln_0_lim[istate] = soln_cell_avg[istate] - M[istate] * pow(this->h, 2.0);
+            }
+
+            a = soln_cell_k[istate] - soln_cell_avg[istate];
+            if (abs(a) <= M[istate] * pow(this->h, 2.0))
+            {
+                soln_k_lim[istate] = soln_cell_avg[istate] + a;
+            }
+            else //if (abs(a) > M * h * h)
+            {
+                if (signbit(a) == signbit(diff_next[istate]) && signbit(a) == signbit(diff_prev[istate]))
+                {
+                    minmod = std::min({ abs(a), abs(diff_next[istate]), abs(diff_prev[istate]) });
+
+                    if (signbit(a))
+                        soln_k_lim[istate] = soln_cell_avg[istate] - std::max(abs(minmod), M[istate] * pow(this->h, 2.0));
+                    else
+                        soln_k_lim[istate] = soln_cell_avg[istate] + std::max(abs(minmod), M[istate] * pow(this->h, 2.0));
+                }
+                else
+                    soln_k_lim[istate] = soln_cell_avg[istate] + M[istate] * pow(this->h, 2.0);
+            }
+
+            real scale = ((soln_cell_0[istate] - soln_cell_avg[istate]) + (soln_cell_k[istate] - soln_cell_avg[istate]));
+            if (scale != 0)
+                theta[istate] = ((soln_0_lim[istate] - soln_cell_avg[istate]) + (soln_k_lim[istate] - soln_cell_avg[istate])) / scale;
+            else
+                theta[istate] = 0;
+        }
+
+        for (unsigned int istate = 0; istate < nstate; ++istate) {
+            for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
+                if (iquad == 0)
+                    soln_at_q[istate][iquad] = soln_0_lim[istate];
+                else if (iquad == n_quad_pts - 1)
+                    soln_at_q[istate][iquad] = soln_k_lim[istate];
+                else
+                {
+                    soln_at_q[istate][iquad] = theta[istate] * (soln_at_q[istate][iquad] - soln_cell_avg[istate])
+                        + soln_cell_avg[istate];
+                }
+            }
```

Comment:
```suggestion
                if (iquad == 0) {
                    soln_at_q[istate][iquad] = soln_0_lim[istate];
                } else if (iquad == n_quad_pts - 1) {
                    soln_at_q[istate][iquad] = soln_k_lim[istate];
                } else {
                    soln_at_q[istate][iquad] = theta[istate] * (soln_at_q[istate][iquad] - soln_cell_avg[istate])
                        + soln_cell_avg[istate];
                }
            }
```

Reply:
I have gone through and modified all if statements and for loops to have consistent format.